src/Lib.P2P/SwarmService.cs

```diff
@@ -0,0 +1,1132 @@
+        {
+            get
+            {
+                return _otherPeers
+                   .Values
+                   .SelectMany(p => p.Addresses);
+            }
+        }
+
+        /// <summary>
+        ///   Get the sequence of all known peers.
+        /// </summary>
+        /// <value>
+        ///   Contains any peer that has been
+        ///   <see cref="RegisterPeerAddress">discovered</see>.
+        /// </value>
+        /// <seealso cref="RegisterPeerAddress"/>
+        public IEnumerable<Peer> KnownPeers => _otherPeers.Values;
+
+        /// <summary>
+        ///   Register that a peer's address has been discovered.
+        /// </summary>
+        /// <param name="address">
+        ///   An address to the peer. It must end with the peer ID.
+        /// </param>
+        /// <returns>
+        ///   The <see cref="Peer"/> that is registered.
+        /// </returns>
+        /// <exception cref="Exception">
+        ///   The <see cref="BlackList"/> or <see cref="WhiteList"/> policies forbid it.
+        ///   Or the "p2p/ipfs" protocol name is missing.
+        /// </exception>
+        /// <remarks>
+        ///   If the <paramref name="address"/> is not already known, then it is
+        ///   added to the <see cref="KnownPeerAddresses"/>.
+        /// </remarks>
+        /// <seealso cref="RegisterPeer(Peer)"/>
+        public Peer RegisterPeerAddress(MultiAddress address)
+        {
+            var peer = new Peer
+            {
+                Id = address.PeerId,
+                Addresses = new List<MultiAddress> {address}
+            };
+
+            return RegisterPeer(peer);
+        }
+
+        /// <summary>
+        ///   Register that a peer has been discovered.
+        /// </summary>
+        /// <param name="peer">
+        ///   The newly discovered peer.
+        /// </param>
+        /// <returns>
+        ///   The registered peer.
+        /// </returns>
+        /// <remarks>
+        ///   If the peer already exists, then the existing peer is updated with supplied
+        ///   information and is then returned.  Otherwise, the <paramref name="peer"/>
+        ///   is added to known peers and is returned.
+        ///   <para>
+        ///   If the peer already exists, then a union of the existing and new addresses
+        ///   is used.  For all other information the <paramref name="peer"/>'s information
+        ///   is used if not <b>null</b>.
+        ///   </para>
+        ///   <para>
+        ///   If peer does not already exist, then the <see cref="PeerDiscovered"/> event
+        ///   is raised.
+        ///   </para>
+        /// </remarks>
+        /// <exception cref="Exception">
+        ///   The <see cref="BlackList"/> or <see cref="WhiteList"/> policies forbid it.
+        /// </exception>
+        public Peer RegisterPeer(Peer peer)
+        {
+            if (peer.Id == null)
+            {
+                throw new ArgumentNullException(nameof(peer));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name '' is not declared in the argument list.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)